oak/proto/application.proto

```diff
@@ -94,6 +94,11 @@ message GrpcClientConfiguration {
   // The endpoint address of the external gRPC service.
   // `address` is represented as an "ip_address:tcp_port" string.
   string address = 1;
+  // Domain name of the external gRPC service.
+  string domain_name = 2;
+  // Loaded PEM encoded X.509 TLS root certificate file used to authenticate an external gRPC
```

Comment:
What happens if the server needs to migrate its issuing CA?  Might need a set of acceptable roots here, so there doesn't have to be a flag day.

Reply:
Will add this as a separate PR, since it will require using raw `rustls` config.
https://github.com/project-oak/oak/issues/965